homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,576 @@
+                        PERCENT_UNITS,
+                        bow["Chlorinator"],
+                    )
+                )
+                sensors.append(
+                    OmniLogicSaltLevelSensor(
+                        coordinator,
+                        "salt_level",
+                        "Salt Level",
+                        backyard,
+                        bow,
+                        None,
+                        "mdi:gauge",
+                        PERCENT_UNITS,
+                        bow["Chlorinator"],
+                    )
+                )
+
+            if "CSAD" in bow:
+                if bow["CSAD"]["systemId"] != "0":
+                    sensors.append(
+                        OmniLogicPHSensor(
+                            coordinator,
+                            "csad_ph",
+                            "pH",
+                            backyard,
+                            bow,
+                            None,
+                            "mdi:gauge",
+                            PERCENT_UNITS,
+                            bow["CSAD"],
+                        )
+                    )
+
+                    sensors.append(
+                        OmniLogicORPSensor(
+                            coordinator,
+                            "csad_orp",
+                            "ORP",
+                            backyard,
+                            bow,
+                            None,
+                            "mdi:gauge",
+                            PERCENT_UNITS,
+                            bow["CSAD"],
+                        )
+                    )
+
+    async_add_entities(sensors, update_before_add=True)
+
+
+class OmnilogicSensor(OmniLogicEntity):
+    """Defines an Omnilogic sensor entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        device_class: str,
+        icon: str,
+        unit: str,
+        sensor_data: dict,
+        system_id: str,
+    ):
+        """Initialize Entities."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            backyard=backyard,
+            bow=bow,
+            icon=icon,
+            entity_data=sensor_data,
+        )
+        self._state = None
+        self._unit_type = backyard["Unit-of-Measurement"]
+        self._device_class = device_class
+        self._unit = unit
+        self._system_id = system_id
+
+    @property
+    def device_class(self):
+        """Return the device class of the entity."""
+        return self._device_class
+
+    @property
+    def unit_of_measurement(self):
+        """Return the right unit of measure."""
+        return self._unit
+
+    @property
+    def force_update(self):
+        """Force update."""
+        return True
+
+    @staticmethod
+    def find_chlorinator(coordinator, systemid):
```

Comment:
If we make this an instance method, we don't need to pass in the `coodinator` parameter. We an access that as `self.coordinator`.

Reply:
Sounds good - addressed.